rfcs/0021-pr-testing.md

```diff
@@ -0,0 +1,85 @@
+---
+feature: pr-testing
+start-date: 2017-11-06
+author: @globin
+co-authors: (find a buddy later to help out with the RFC)
+related-issues: https://github.com/NixOS/hydra/pull/516 https://github.com/mayflower/nixborg
+---
+
+# Summary
+[summary]: #summary
+
+In order to properly test Pull Requests and reduce the maintainers' workload
+we need automation to build PRs on Hydra. This RFC specifies an architecture
+to achieve this and contains a reference implementation.
+
+# Motivation
+[motivation]: #motivation
+
+We currently have peaks of ~70 PRs a month at release time, this consumes an
+immense amount of maintainers' time to test the PRs.
+
+Another issue is the difficulty of testing mass-rebuilds. Only a few people
+have resources for themselves to test such computation intensive builds and
+Hydra access is limited to only a small amount of people.
+
+Due to having already built all packages prior to merging this would reduce
+the need for staging.
+
+# Detailed design
+[design]: #detailed-design
+
+The general idea is to have a service which receives webhooks from github,
+then requires some acknowledgement of a maintainer to build/test a PR and
```

Comment:
Re: «a maintainer» — maybe add something about supporting a flexible definition of a maintainer (any NixOS project member/Nixpkgs committer or the package maintainer who doesn't yet have access) to «Future work»? Not in the main body, because this discussion should not delay checking if Hydra can build all PRs.

Reply:
See #19 also.

AFAIK the current implementation of that bot just has a hardcoded list of people who should be allowed access.

I think that people from the maintainers file + anyone who currently has commit access should be allowed to instruct the bot via @-mentions.

Reply:
Currently "a maintainer" refers to a contributor with commit access, "member" in GitHub terminology, I'll clarify this.